src/components/feedback/ModalDialog.tsx

```diff
@@ -0,0 +1,74 @@
+import classnames from 'classnames';
+
+import { useSyncedRef } from '../../hooks/use-synced-ref';
+import { downcastRef } from '../../util/typing';
+import Overlay from '../layout/Overlay';
+import Dialog from './Dialog';
+import type { DialogProps } from './Dialog';
+
+type ComponentProps = {
+  size?: 'sm' | 'md' | 'lg' | 'custom';
+};
+
+export type ModalDialogProps = DialogProps & ComponentProps;
+
+/**
+ * Show a modal dialog
+ */
+const ModalDialogNext = function ModalDialog({
```

Comment:
Does it make sense to continue using the `Next` suffix for components which don't have a non-next equivalent?

Reply:
My current plan is to do this in one go when working on v6 of the package; it's a task tracked here: https://github.com/hypothesis/frontend-shared/issues/876

Until then, it's a marker of a non-deprecated component.